go/cap/cap_test.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2018 The cap Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cap
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func getCAPAlertExample() (*Alert, error) {
+	xmlData, err := ioutil.ReadFile("../../resources/cap_amber_alert_example.xml")
+	if err != nil {
+		return nil, err
+	}
+	return ParseAlert(xmlData)
+}
+
+func TestUnmarshalAlertHasProperValues(t *testing.T) {
+	alert, err := getCAPAlertExample()
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, alert.Identifier, "KAR0-0306112239-SW")
+	assert.Equal(t, alert.Sender, "KARO@CLETS.DOJ.CA.GOV")
+	assert.Equal(t, string(alert.Sent), "2003-06-11T22:39:00-07:00")
+	assert.Equal(t, alert.Status, "Actual")
+	assert.Equal(t, alert.MsgType, "Alert")
+	assert.Equal(t, alert.Scope, "Public")
+	assert.Equal(t, alert.Note, "")
+	assert.Equal(t, len(alert.Info), 2)
+}
+
+func TestUnmarshalAlertInfoHasProperValues(t *testing.T) {
+	alert, err := getCAPAlertExample()
+	if err != nil {
+		t.Fatal(err)
+	}
+	var info = alert.Info[0]
+	assert.Equal(t, info.Category[0], "Rescue")
+	assert.Equal(t, info.Event, "Child Abduction")
+	assert.Equal(t, info.Urgency, "Immediate")
+	assert.Equal(t, info.Certainty, "Likely")
+	assert.Equal(t, info.EventCode[0].ValueName, "SAME")
+	assert.Equal(t, info.EventCode[0].Value, "CAE")
+	assert.Equal(t, string(info.Effective), "")
+	assert.Equal(t, string(info.Expires), "")
+	assert.Equal(t, info.SenderName, "Los Angeles Police Dept - LAPD")
+	assert.Equal(t, info.Headline, "Amber Alert in Los Angeles County")
+	assert.Contains(t, info.Description, "DATE/TIME: 06/11/03, 1915 HRS.  VICTIM(S): KHAYRI D")
+	assert.Contains(t, info.Instruction, "")
+	assert.Equal(t, len(info.Parameter), 0)
+	assert.Equal(t, len(info.Area), 1)
```

Comment:
it'd be cool to see this data converted to example files that can be reused among other impls later. May need a separate repo. non-blocking now.

Reply:
yeah we keep talking about creating a decent reusable test_infra ... I worked on this one and bring it around with me...

Reply:
Example files or a validator. Have each impl send the original contents and parsed json / yaml over to the validator who would do it's own json conversion and then do a deep comparison. Just a thought. Not married to it.